pkg/controller/elasticsearch/certificates/transport/reconcile.go

```diff
@@ -74,11 +76,14 @@ func DeleteStatefulSetTransportCertificate(client k8s.Client, namespace string,
 
 // DeleteLegacyTransportCertificate ensures that the former Secret which used to contain the transport certificates is deleted.
 func DeleteLegacyTransportCertificate(client k8s.Client, es esv1.Elasticsearch) error {
-	secret := corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: es.Namespace,
-			Name:      esv1.LegacyTransportCertsSecretSuffix(es.Name),
-		},
+	var secret corev1.Secret
+	// do a GET from the cache first before attempting a DElETE that hits the API server
+	err := client.Get(context.Background(), types.NamespacedName{Namespace: es.Namespace, Name: esv1.LegacyTransportCertsSecretSuffix(es.Name)}, &secret)
```

Comment:
Question, not an issue, but why do we use `context.Background()` throughout the codebase for kubernetes client.* calls instead of something with a default timeout?

Reply:
I think that's a valid question maybe raise a separate issue to discuss it though?

Reply:
FWIW https://github.com/elastic/cloud-on-k8s/issues/5450 indicates that a 1m timeout is applied to all requests